### PR TITLE
[Feature] support run export in sync mode

### PR DESCRIPTION
**Pull request by hoffermei**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [x] Interface/UI changes: syntax, type conversion, expression evaluation, display information

Currenttly run export command always in async mode, user should `select last_query_id()` and then check execute result by `show export where queryid='{queryId}'`, this pr add `sync` after origin command make it run in sync mode and result in final execute result when export finished or cancelled. return user with final show export result.

```sql
/* difference: add sync at end of export comand */
mysql> EXPORT TABLE olap_tbl_02 PARTITION (p20220418) TO "s3a://hoffer/tmp/" properties (     
  "load_mem_limit"="21474836480",     
  "bytes_per_file" = "123731968" 
) WITH BROKER (     
  "fs.s3a.access.key" = "xxxxx",     
  "fs.s3a.secret.key" = "yyyyy",      
  "fs.s3a.connection.ssl.enabled" = "false",     
   "fs.s3a.signing-algorithm" = "S3SignerType",      
   "fs.s3a.etag.checksum" = "false",      
   "fs.s3a.endpoint" = "cos.ap-guangzhou.tencentcos.cn" 
 ) sync \G
*************************** 1. row ***************************
     JobId: 561118
   QueryId: af7cbd51-4636-11ee-b324-02422fe84e07
     State: FINISHED
  Progress: 100%
  TaskInfo: {"partitions":["p20220418"],"column separator":"\t","columns":["*"],"broker":"","row delimiter":"\n","mem limit":21474836480,"tablet num":1,"where":"","coord num":2,"db":"hoffertest","tbl":"olap_tbl_02"}
      Path: s3a://hoffer/temp/
CreateTime: 2023-08-29 14:38:38
 StartTime: 2023-08-29 14:38:39
FinishTime: 2023-08-29 14:38:48
   Timeout: 86400
  ErrorMsg: NULL
1 row in set (15.01 sec)
```

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 1]
```diff
@@ -1723,8 +1723,12 @@ showWhiteListStatement
 
 // ------------------------------------------- Export Statement --------------------------------------------------------
 
+asyncStatement
+    : SYNC | ASYNC
+    ;
+
 exportStatement
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:1730]: It's better to add a session variable like `sync_mode`. Adding `SYNC|ASYNC` in the command suffix is strange.

Comment (hoffermei) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:1726]: has changed to use `(WITH (SYNC | ASYNC) MODE)?` like [ANALYZE TABLE](https://docs.starrocks.io/zh-cn/latest/sql-reference/sql-statements/data-definition/ANALYZE%20TABLE#%E8%AF%AD%E6%B3%95) and [REFRESH MATERIALIZED VIEW](https://docs.starrocks.io/zh-cn/latest/sql-reference/sql-statements/data-manipulation/REFRESH%20MATERIALIZED%20VIEW#%E8%AF%AD%E6%B3%95)

**Review by chaoyli — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java [hunk 2]
```diff
@@ -1450,6 +1452,45 @@ private void handleExportStmt(UUID queryId) throws Exception {
...
+            return;
+        }
+        // poll and check export job state
+        while (true) {
+            ExportJob exportJob = context.getGlobalStateMgr().getExportMgr().getExportByQueryId(queryId);
+            if (exportJob == null ||
+                    (exportJob.getState() != ExportJob.JobState.FINISHED
+                    && exportJob.getState() != ExportJob.JobState.CANCELLED)) {
+                try {
+                    Thread.sleep(Config.export_job_sync_poll_interval_seconds * 1000);
```
Comment (Astralidea) [inline fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:1466]: maybe you don't need add more config ? just use export_checker_interval_second?

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java [hunk 3]
```diff
@@ -1450,6 +1452,45 @@ private void handleExportStmt(UUID queryId) throws Exception {
         ExportStmt exportStmt = (ExportStmt) parsedStmt;
         exportStmt.setExportStartTime(context.getStartTime());
         context.getGlobalStateMgr().getExportMgr().addExportJob(queryId, exportStmt);
+
+        if (!exportStmt.getSync()) {
+            return;
+        }
+        // poll and check export job state
+        while (true) {
```
Comment (Astralidea) [inline fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:1460]: It is better not to use infinite loop，maybe you can use export_task_default_timeout_second + add a little more time

**Review by Astralidea — DISMISSED** [review]

**Review by chaoyli — DISMISSED** [review]

**Review by Astralidea — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 7 / 45 (15.56%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/ExportMgr.java|0|9|00.00%|[183, 184, 186, 188, 189, 190, 192, 194, 196]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|2|30|06.67%|[1532, 1533, 1534, 1535, 1536, 1537, 1539, 1540, 1541, 1543, 1544, 1545, 1546, 1547, 1553, 1554, 1555, 1556, 1557, 1558, 1559, 1560, 1562, 1564, 1565, 1566, 1567, 1570]|
|:large_blue_circle:|com/starrocks/sql/ast/ExportStmt.java|5|6|83.33%|[113]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by chaoyli — DISMISSED** [review]

**Review by chaoyli — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 37 / 45 (82.22%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|23|30|76.67%|[1544, 1545, 1547, 1562, 1565, 1566, 1567]|
|:large_blue_circle:|com/starrocks/sql/ast/ExportStmt.java|5|6|83.33%|[113]|
|:large_blue_circle:|com/starrocks/load/ExportMgr.java|9|9|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Pull request merged**